src/zenml/models/v2/base/filter.py

```diff
@@ -190,6 +199,40 @@ def generate_query_conditions_from_column(self, column: Any) -> Any:
             return column.endswith(f"{self.value}")
         if self.operation == GenericFilterOps.NOT_EQUALS:
             return column != self.value
+        if self.operation == GenericFilterOps.ONEOF:
```

Comment:
I added a pydantic `after` validator to do the check. 

Btw, we do the same check within `_resolve_operator` and `_define_uuid_filter` as well. It is a similar pattern for the other types too. Perhaps, it would be nice to optimize this logic.